modules/ROOT/pages/dataweave-indexed-readers.adoc

```diff
@@ -0,0 +1,51 @@
+= Indexed Readers in DataWeave
+
+DataWeave can read input data (such as XML, JSON, and CSV) in an indexed strategy to avoid loading the whole document in memory while still allowing random access just like in the in-memory strategy. Use indexed readers when the input is larger than a certain configurable threshold or if the `indexedReader` setting is set to `true`.
+
+To manage files that don’t fit in memory, write the files to a temporary file in disk and index the document content beforehand, which also takes time and disk space.
+
+Indexed readers can process input files up to 20GBs. For bigger files use the streaming mode, which is faster. Despite this mode being more limited in functionality, there’s no maximum input size limitation for streaming readers.
+
+The actual maximum input file size supported is difficult to estimate because it depends on the content of the input.
+
+The following list shows the limits on different parts of the input document:
```

Comment:
Are these limits that apply to all input? Should it precede the text about using indexed readers, if they are the solution to these limits?
I don't understand the narrative flow of this topic.